gui/src/3D/controls/nearMenu.ts

```diff
@@ -0,0 +1,203 @@
+import { Scene } from "babylonjs/scene";
+import { Vector3 } from "babylonjs/Maths/math.vector";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Nullable } from "babylonjs/types";
+import { Control3D } from "./control3D";
+import { VolumeBasedPanel } from "./volumeBasedPanel";
+import { Mesh } from "babylonjs/Meshes/mesh";
+import { BoxBuilder } from "babylonjs/Meshes/Builders/boxBuilder";
+import { AbstractMesh } from "babylonjs/Meshes/abstractMesh";
+import { FluentMaterial } from "../materials/fluent/fluentMaterial";
+import { Color3 } from "babylonjs/Maths/math.color";
+import { Observer } from "babylonjs/Misc/observable";
+import { Logger } from "babylonjs/Misc/logger";
+import { Container3D } from "./container3D";
+import { TouchHolographicButton } from "./touchHolographicButton";
+import { DefaultBehavior } from "../behaviors/defaultBehavior";
+import { StandardMaterial } from "babylonjs/Materials/standardMaterial";
+
+/**
+ * NearMenu that displays buttons and follows the camera
+ */
+export class NearMenu extends VolumeBasedPanel {
+    /**
+     * Base Url for the assets.
+     */
+    private static ASSETS_BASE_URL: string = "https://assets.babylonjs.com/meshes/MRTK/";
+    /**
+     * File name for the close icon.
+     */
+    private static PIN_ICON_FILENAME: string = "IconPin.png";
+
+    private _pinButton: TouchHolographicButton;
+    private _backPlate: Mesh;
+    private _backPlateMaterial: FluentMaterial;
+    private _pinMaterial: StandardMaterial;
+    private _pickedPointObserver: Nullable<Observer<Nullable<Vector3>>>;
+    private _defaultBehavior: DefaultBehavior;
+
+    private _currentMin: Nullable<Vector3>;
+    private _currentMax: Nullable<Vector3>;
+
+    private _isPinned: boolean = false;
+    /**
+     * Indicates if the near menu is world-pinned
+     */
+    public get isPinned(): boolean {
+        return this._isPinned;
+    }
+
+    public set isPinned(value: boolean) {
+        this._isPinned = value;
+
+        if (this._isPinned) {
+            this._pinMaterial.emissiveColor.copyFromFloats(0.25, 0.4, 0.95);

```

Comment:
Should the user be able to update those colors (see below too)?

Reply:
This is really temp code as a new shader is coming to replace the FluentMaterial. There is also the same hard coded color in the GUI slate.

Reply:
https://github.com/BabylonJS/Babylon.js/pull/10397 <- waiting for this PR